From 359459d85fd388a03a8cd0e43389759dd4b79f3d Mon Sep 17 00:00:00 2001
From: "Wang, ArvinX" <arvinx.wang@intel.com>
Date: Thu, 22 Aug 2019 14:22:46 +0800
Subject: [PATCH] Post bugreport notification and activity to user 10

Grand user 10 to access to user 0 bugreport

Change-Id: Id19c39d0fdb38c73cfdf221dc8a6e3b407f8ee53
Tracked-On: OAM-85032
Signed-off-by: Wang, ArvinX <arvinx.wang@intel.com>
---
 packages/Shell/AndroidManifest.xml                        | 2 ++
 .../src/com/android/shell/BugreportProgressService.java   | 8 ++++----
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/packages/Shell/AndroidManifest.xml b/packages/Shell/AndroidManifest.xml
index da139d74f8a..1fe6fe1b5c1 100644
--- a/packages/Shell/AndroidManifest.xml
+++ b/packages/Shell/AndroidManifest.xml
@@ -208,11 +208,13 @@
     <application android:label="@string/app_label"
                 android:theme="@android:style/Theme.DeviceDefault.DayNight"
                 android:defaultToDeviceProtectedStorage="true"
+                android:persistent="true"
                 android:directBootAware="true">
         <provider
             android:name="androidx.core.content.FileProvider"
             android:authorities="com.android.shell"
             android:grantUriPermissions="true"
+            android:singleUser="true"
             android:exported="false">
             <meta-data
                 android:name="android.support.FILE_PROVIDER_PATHS"
diff --git a/packages/Shell/src/com/android/shell/BugreportProgressService.java b/packages/Shell/src/com/android/shell/BugreportProgressService.java
index e57a5ded069..a4c25b71498 100644
--- a/packages/Shell/src/com/android/shell/BugreportProgressService.java
+++ b/packages/Shell/src/com/android/shell/BugreportProgressService.java
@@ -550,7 +550,7 @@ public class BugreportProgressService extends Service {
     private void sendForegroundabledNotification(int id, Notification notification) {
         if (mForegroundId >= 0) {
             if (DEBUG) Log.d(TAG, "Already running as foreground service");
-            NotificationManager.from(mContext).notify(id, notification);
+            NotificationManager.from(mContext).notifyAsUser(null, id, notification, UserHandle.CURRENT);
         } else {
             mForegroundId = id;
             Log.d(TAG, "Start running as foreground service on id " + mForegroundId);
@@ -991,7 +991,7 @@ public class BugreportProgressService extends Service {
         if (useChooser) {
             sendShareIntent(mContext, notifIntent);
         } else {
-            mContext.startActivity(notifIntent);
+            mContext.startActivityAsUser(notifIntent, UserHandle.CURRENT);
         }
 
         // ... and stop watching this process.
@@ -1007,7 +1007,7 @@ public class BugreportProgressService extends Service {
         chooserIntent.putExtra(ChooserActivity.EXTRA_PRIVATE_RETAIN_IN_ON_STOP, true);
         // Starting the activity from a service.
         chooserIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-        context.startActivity(chooserIntent);
+        context.startActivityAsUser(chooserIntent, UserHandle.CURRENT);
     }
 
     /**
@@ -1051,7 +1051,7 @@ public class BugreportProgressService extends Service {
         }
 
         Log.v(TAG, "Sending 'Share' notification for ID " + info.id + ": " + title);
-        NotificationManager.from(mContext).notify(info.id, builder.build());
+        NotificationManager.from(mContext).notifyAsUser(null, info.id, builder.build(), UserHandle.CURRENT);
     }
 
     /**
-- 
2.21.0

